src/commit.c

```diff
@@ -104,7 +104,7 @@ int git_commit_create_from_callback(
 
 	if (update_ref != NULL) {
 		error = git_reference__update_for_commit(
-			repo, ref, update_ref, id, committer, "commit");
```

Comment:
We do use the commiter, but we read it from the commit object itself, which we load in `__update_for_commit()`.